### PR TITLE
Tidy up

### PR DESCRIPTION
**Pull request by daviddias**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,17 +57,57 @@ go get github.com/libp2p/go-libp2p-pubsub
...
+├── pubsub.go
+├── sign.go
+├── sign_test.go
+├── subscription.go
+├── topic.go
+├── topic_test.go
+├── trace.go
+├── tracer.go
+└── validation.go
+```
```
Comment (daviddias) [inline README.md]: @vyzo did I get this separation right?

Comment (daviddias) [reply README.md]: PS: I would like to break them into separate folders post https://github.com/libp2p/go-libp2p-pubsub/pull/263 is merged

Comment (vyzo) [reply README.md]: no, blacklist, comm, discovery, and notify are used by all, they are not gossipsub specific.
The only thing that is gossipsub specific is `gossipsub.go` and `mcache.go`.

Comment (vyzo) [reply README.md]: Also, we can't break into separate folders without major refactoring, the routers need access to pubsub protected members.

Comment (vyzo) [reply README.md]: plus you shouldn't put the test files in there, they are irrelevant for presentation purposes.

Comment (daviddias) [reply README.md]: Thank you! Fair points! Updated

File: README.md [hunk 2]
```diff
@@ -34,23 +57,54 @@ go get github.com/libp2p/go-libp2p-pubsub
...
-See [libp2p/specs/pubsub#Implementations](https://github.com/libp2p/specs/tree/master/pubsub#Implementations).
+`To be written`
 
 ## Documentation
 
-See the [libp2p specs](https://github.com/libp2p/specs/tree/master/pubsub) for high level documentation
-and [godoc](https://godoc.org/github.com/libp2p/go-libp2p-pubsub) for API documentation.
+See the [libp2p specs](https://github.com/libp2p/specs/tree/master/pubsub) for high level documentation and [godoc](https://godoc.org/github.com/libp2p/go-libp2p-pubsub) for API documentation.
 
+### In this repo, you will find
```
Comment (vyzo) [inline README.md:68]: Is this really useful?

Comment (daviddias) [reply README.md:68]: I believe it is. Have used it in other projects to lay the map of the land for newcomers. It reduces the "daunting factor" when arriving at a unknown codebase

Comment (vyzo) [reply README.md:68]: I would order the subsystem implementation files first and the routers last.

File: README.md [hunk 3]
```diff
@@ -34,23 +57,54 @@ go get github.com/libp2p/go-libp2p-pubsub
...
+.
+├── LICENSE
+├── README.md
+# Regular Golang repo set up
+├── codecov.yml
+├── pb
+├── go.mod
+├── go.sum
+├── doc.go
+# Floodsub specific
```
Comment (vyzo) [inline README.md]: Floodsub router implementation

File: README.md [hunk 4]
```diff
@@ -34,23 +57,54 @@ go get github.com/libp2p/go-libp2p-pubsub
...
+├── floodsub.go
+# Randomsub specific
```
Comment (vyzo) [inline README.md]: Randomsub router implementation

File: README.md [hunk 5]
```diff
@@ -34,23 +57,54 @@ go get github.com/libp2p/go-libp2p-pubsub
...
+├── randomsub.go
+# Gossipsub specific
```
Comment (vyzo) [inline README.md]: Gossipsub router implementation

File: README.md [hunk 6]
```diff
@@ -34,23 +57,54 @@ go get github.com/libp2p/go-libp2p-pubsub
...
+├── gossipsub.go
+├── mcache.go
+# Used by all
```
Comment (vyzo) [inline README.md]: Pubsub subsystem base

**Review by vyzo — COMMENTED** [review]
Let's be a little more descriptive and not say "used by all", it's not accurate.

**Comment by daviddias** [comment]
Done, thanks for the review, @vyzo

**Review by vyzo — APPROVED** [review]

**Pull request merged**